### PR TITLE
datatree accessor to multiscale spatial image

### PR DESCRIPTION
**Pull request by giovp**
This is a quick try to use the xarray accessors as suggested in #80 , changes are minimal. 
As suggested a `MultiscaleSpatialImage` can be identified with:
```python
isinstance(i, DataTree) and hasattr(i, 'msi')
```
or 
```python
isinstance(i, DataTree) and isinstance(i.msi, MultiscaleSpatialImage)
```

wdyt @thewtex ?

p.s. tests pass locally...

### CODE REVIEW

File: multiscale_spatial_image/multiscale_spatial_image.py [hunk 1]
```diff
@@ -24,17 +27,16 @@ class MultiscaleSpatialImage(DataTree):
       scale2
     """
 
-    def __init__(
+    def __init__(self, xarray_obj: DataTree):
+        self._obj = xarray_obj
```
Comment (thewtex) [inline multiscale_spatial_image/multiscale_spatial_image.py:31]: Perhaps something more specific than `_obj`, `_dt`, etc.

**Review by thewtex — COMMENTED** [review]
@giovp awesome!!

I will work on fixing unrelated CI failures in another PR.

**Comment by thewtex** [comment]
ci is green on `main` -- please rebase

File: multiscale_spatial_image/multiscale_spatial_image.py (hunk 1 above)
Comment (giovp) [reply multiscale_spatial_image/multiscale_spatial_image.py]: changed to `_dt`

File: test/test_ngff_validation.py [hunk 2]
```diff
@@ -23,9 +25,10 @@ def load_schema(version: str = "0.4", strict: bool = False) -> Dict:
     schema = json.loads(response.data.decode())
     return schema
 
-def check_valid_ngff(multiscale: MultiscaleSpatialImage):
+def check_valid_ngff(multiscale: DataTree):
     store = zarr.storage.MemoryStore(dimension_separator="/")
-    multiscale.to_zarr(store, compute=True)
+    assert isinstance(multiscale.msi, MultiscaleSpatialImage)
+    multiscale.msi.to_zarr(store, compute=True)
```
Comment (thewtex) [inline test/test_ngff_validation.py:31]: Nice!

**Review by thewtex — APPROVED** [review]
LGTM!

**Pull request merged**